main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -1016,39 +1014,44 @@ object Weeder extends Phase[ParsedAst.Program, WeededAst.Program] {
       }
 
     case ParsedAst.Expression.Interpolation(sp1, parts, sp2) =>
-      val loc = mkSL(sp1, sp2)
 
-      parts match {
-        case Seq() =>
-          // Case 1: Return the empty string if there are no interpolation parts.
-          WeededAst.Expression.Str("", loc).toSuccess
+      /**
+        * Returns an expression that concatenates the result of the expression `e1` with the expression `e2`.
+        */
+      def mkConcat(e1: WeededAst.Expression, e2: WeededAst.Expression, loc: SourceLocation): WeededAst.Expression = {
+        val op = BinaryOperator.Plus
+        WeededAst.Expression.Binary(op, e1, e2, loc)
+      }
 
-        case Seq(ParsedAst.InterpolationPart.ExpPart(e)) =>
-          // Case 2: Return the expression if there is only a single expression.
-          visitExp(e)
+      /**
+        * Returns an expression that applies `toString` to the result of the given expression `e`.
+        */
+      def mkApplyShow(e: WeededAst.Expression, sp1: SourcePosition, sp2: SourcePosition): WeededAst.Expression = {
+        val fqn = "ToString.toString"
+        mkApplyFqn(fqn, List(e), sp1, sp2)
+      }
 
-        case Seq(ParsedAst.InterpolationPart.StrPart(s)) =>
-          // Case 3: Return the string if there is only a single string.
-          WeededAst.Expression.Str(s, loc).toSuccess
+      val loc = mkSL(sp1, sp2)
 
-        case ps =>
-          // Case 4: Construct a sequence of string append expressions.
+      parts match {
+        case Seq(ParsedAst.InterpolationPart.StrPart(innerSp1, lit, innerSp2)) =>
+          // Special case: We have a constant string. Simply return it.
+          WeededAst.Expression.Str(lit, mkSL(innerSp1, innerSp2)).toSuccess
```

Comment:
Does this indicate a likely bug that we should report?

Reply:
No, this is just an artifact of the fact that `"abc"` is a string interpolator without anything in it.